tools/clang/unittests/HLSLTestLib/FileCheckerTest.cpp

```diff
@@ -409,8 +409,13 @@ FileRunCommandResult FileRunCommandPart::RunDxcHashTest(dxc::DxcDllSupport &DllS
   CComPtr<IDxcBlob> pOriginalHash;
   // If failed the original compilation, just pass the test. The original test was likely
   // testing for failure.
-  if (FAILED(CompileForHash(opts, CommandFileName, DllSupport, original_flags, &pOriginalHash, originalOutput)))
+  try {
+    if (FAILED(CompileForHash(opts, CommandFileName, DllSupport, original_flags,
+                              &pOriginalHash, originalOutput)))
+      return FileRunCommandResult::Success();
+  } catch (...) {
```

Comment:
I don't think exceptions from test should be translated to success here.  Exceptions at this point indicate broken test infrastructure or bug in API, not the kind of failure condition that a test might be expecting.